backend/src/main/java/eu/deltasource/elearning/service/InstructorService.java

```
@@ -0,0 +1,83 @@
+package eu.deltasource.elearning.service;
+
+import eu.deltasource.elearning.DTOs.InstructorDTO;
+import eu.deltasource.elearning.exception.InstructorNotFoundException;
+import eu.deltasource.elearning.model.Instructor;
+import eu.deltasource.elearning.repository.InstructorRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Optional;
+
+/**
+ * Service class for managing instructor-related operations.
+ * This class acts as an intermediary between the controller and the repository,
+ * handling business logic for instructor management.
+ */
+@RequiredArgsConstructor
+@Service
+public class InstructorService {
+    private final InstructorRepository instructorRepository;
+
+    @Transactional
+    public InstructorDTO createInstructor(InstructorDTO instructorDTO) {
+        Instructor instructor = mapToInstructor(instructorDTO);
+        Optional<Instructor> existingInstructor = instructorRepository.findByEmail(instructor.getEmail());
+        if(existingInstructor.isPresent()){
+            return mapToInstructorDTO(existingInstructor.get());
+        }
+        instructor = instructorRepository.save(instructor);
+        return mapToInstructorDTO(instructor);
+    }
```

I believe that if the instructor already exists, we should throw an exception rather than returning the existing one.